lib/ask/runtime/session.ex

```diff
@@ -673,10 +681,43 @@ defmodule Ask.Runtime.Session do
     end)
   end
 
-  defp falls_in_quota_already_completed?(respondent, flow) do
+  # Must the respondent be rejected?
+  defp must_be_rejected?(
+         respondent,
+         reply_disposition,
+         in_quota_completed_steps,
+         count_partial_results
+       ) do
+    # The respondent disposition is updated by Runtime.Survey after the current step is handled
+    # by Runtime.Session, but the quota bucket could be assigned to the respondent and the quota
+    # incremented before the disposition update happens. So if the disposition should be updated
+    # the respondent also must be tested using their new disposition.
+    respondent_incremented_their_quota? =
+      if Flow.should_update_disposition(respondent.disposition, reply_disposition) do
+        updated_respondent = %{respondent | disposition: reply_disposition}
+
+        Respondent.incremented_their_quota?(respondent, count_partial_results) ||
+          Respondent.incremented_their_quota?(updated_respondent, count_partial_results)
+      else
+        Respondent.incremented_their_quota?(respondent, count_partial_results)
+      end
+
     cond do
-      flow.in_quota_completed_steps -> false
-      respondent.quota_bucket_id == nil -> false
+      # Was the respondent already rejected?
+      # flow.in_quota_completed_steps ~= (respondent.disposition == "rejected)
```

Comment:
what does this comment mean?

Reply:
I was trying to give the reader the clue that asking for `flow.in_quota_completed_steps` is pretty much the same as asking for `respondent.disposition == "rejected"` but I removed it because I see it looked rare.